### ISSUE TITLE
Timeout error on rasa shell

### ISSUE THREAD

**Issue by tabergma** [RasaHQ/rasa#4606]
**Rasa version**: 1.3.9

**Python version**: 3.6

**Issue**:
https://forum.rasa.com/t/timeout-error-on-rasa-shell/19638

Error can be reproduced on `examples/restaurantbot`. Add a `time.sleep(15)` to any action and run the bot.

**Error (including full traceback)**:
```
2019-10-14 10:18:37 ERROR    asyncio  - Task exception was never retrieved
future: <Task finished coro=<configure_app.<locals>.run_cmdline_io() done, defined at /Users/tabergma/.virtualenv/rasa/lib/python3.6/site-packages/rasa/core/run.py:123> exception=TimeoutError()>
Traceback (most recent call last):
  File "/Users/tabergma/.virtualenv/rasa/lib/python3.6/site-packages/rasa/core/run.py", line 127, in run_cmdline_io
    server_url=constants.DEFAULT_SERVER_FORMAT.format("http", port)
  File "/Users/tabergma/.virtualenv/rasa/lib/python3.6/site-packages/rasa/core/channels/console.py", line 138, in record_messages
    async for response in bot_responses:
  File "/Users/tabergma/.virtualenv/rasa/lib/python3.6/site-packages/async_generator/_impl.py", line 366, in step
    return await ANextIter(self._it, start_fn, *args)
  File "/Users/tabergma/.virtualenv/rasa/lib/python3.6/site-packages/async_generator/_impl.py", line 205, in throw
    return self._invoke(self._it.throw, type, value, traceback)
  File "/Users/tabergma/.virtualenv/rasa/lib/python3.6/site-packages/async_generator/_impl.py", line 209, in _invoke
    result = fn(*args)
  File "/Users/tabergma/.virtualenv/rasa/lib/python3.6/site-packages/rasa/core/channels/console.py", line 103, in send_message_receive_stream
    async for line in resp.content:
  File "/Users/tabergma/.virtualenv/rasa/lib/python3.6/site-packages/aiohttp/streams.py", line 40, in __anext__
    rv = await self.read_func()
  File "/Users/tabergma/.virtualenv/rasa/lib/python3.6/site-packages/aiohttp/streams.py", line 329, in readline
    await self._wait('readline')
  File "/Users/tabergma/.virtualenv/rasa/lib/python3.6/site-packages/aiohttp/streams.py", line 297, in _wait
    await waiter
  File "/Users/tabergma/.virtualenv/rasa/lib/python3.6/site-packages/aiohttp/helpers.py", line 585, in __exit__
    raise asyncio.TimeoutError from None
concurrent.futures._base.TimeoutError
```

**Command or request that led to error**:
```
rasa shell
```

**Comment by ravikrcs** [comment]
same problem on
Rasa version : 1.5.1
python 3.6.8

**Comment by wochinge** [comment]
This is due to https://github.com/RasaHQ/rasa/blob/f950f7c734e2482fd49bd97fb5a955d0708a1287/rasa/core/channels/console.py#L101 . We are doing this to avoid freezing of the shell when the rasa server crashes in background. 

For me the question is, why is the action running so long?

**Comment by ravikrcs** [comment]
Please check this [thread](https://forum.rasa.com/t/asyncio-timeout-error-task-exception-was-never-retrieved/22342), in which i am calling an action for http connection.

**Comment by tdplaza** [comment]
It' because default time out (10 seconds) is not enough for your action to be executed. 

Increase the default timeout in "/rasa/core/channels/console.py" file, line 22 might help solving problem. Or you can check which makes the action taking time as @wochinge comment and optimize the executing time.

**Comment by ravikrcs** [comment]
This is my action file that gives timeout error

![human_action](https://user-images.githubusercontent.com/31586794/71237096-9ee8ae80-2326-11ea-8a57-8f380092ee0d.PNG)

**Comment by ravikrcs** [comment]
Here is my action file that gives timeout error:

![gitaction](https://user-images.githubusercontent.com/31586794/71237455-8f1d9a00-2327-11ea-89d7-4a15db487a2f.PNG)

**Comment by wochinge** [comment]
@ravikrcs Can you mock the human handoff server for the `rasa shell` tests so it returns sth immediately?

**Comment by ravikrcs** [comment]
could you rephrase? 

This is my api.py file

![api](https://user-images.githubusercontent.com/31586794/72318811-59b86200-36c3-11ea-8959-07cbdd56a7a6.PNG)

i am testing through Postman. But
It doesn't return any response.

**Comment by ravikrcs** [comment]
getting this error on rasa shell:
![error01](https://user-images.githubusercontent.com/31586794/72319319-e1eb3700-36c4-11ea-9f5b-84aa31a55fcd.PNG)

**Comment by wochinge** [comment]
> could you rephrase?

I meant if you can replace the handoff interaction for testing the bot with something which returns an answer immediately? As a workaround you could also use the `rest` input channel for testing (in case you want to manually answer to the handoff requests)

**Comment by ravikrcs** [comment]
> I meant if you can replace the handoff interaction for testing the bot with something which returns an answer immediately? 

Yes @wochinge ,
other action class are working fine. only problem in human-handoff class that gives timeout error that means i am not getting response from api which i mentioned above.

**Comment by ravikrcs** [comment]
i got this issue in human - handoff implementation. Below i attached screenshots.....

actions.py
![actions](https://user-images.githubusercontent.com/31586794/72497550-cd3ca980-3852-11ea-9bc4-940552f3788a.PNG)

api.py
![api](https://user-images.githubusercontent.com/31586794/72497594-e2193d00-3852-11ea-9f94-6bfa708ff2c1.PNG)

action server:
![action_server](https://user-images.githubusercontent.com/31586794/72497619-f8bf9400-3852-11ea-8e60-507d01358091.PNG)


flask server:
![flask](https://user-images.githubusercontent.com/31586794/72497645-0a08a080-3853-11ea-85f1-70c289cb5c5f.PNG)


rasa shell:
![shell_01](https://user-images.githubusercontent.com/31586794/72497661-1a208000-3853-11ea-8a0c-b61a2c145f56.PNG)
![shell_02](https://user-images.githubusercontent.com/31586794/72497666-1ee53400-3853-11ea-8f22-6279f67c512a.PNG)

**Comment by wochinge** [comment]
mhm, ok, then let's 
- check if there is a better way to communicate errors in Rasa to the `console` connector than the timeout
- if not: make the timeout configurable

**Comment by mtrworld** [comment]
Is there any update or workaround for this issue?

**Comment by wochinge** [comment]
A workaround for now would be to not use `rasa shell` but e.g. Rasa X or the rest api

**Comment by rbasto1** [comment]
I'm having the same issue. We use `rasa shell` in development in order to have a lighter environment (just two containers needed, rasa-worker and the actions app).
When we load a new model the API call to `PUT /model` takes more than 10 seconds, and the console crashes.
Can you please add a config variable that overrides `DEFAULT_STREAM_READING_TIMEOUT_IN_SECONDS`?
Thanks!

**Comment by jutinguely** [comment]
I get the same issue here! Any update?

**Comment by wochinge** [comment]
@jutinguely This is still on our todo list. If you want, you can also pick it up? I'd be happy to give you feedback / support you on the way.

**Comment by TKueng** [comment]
Did you already do that? Because I could adjust the default stream time to 25 seconds in console.py file:
`DEFAULT_STREAM_READING_TIMEOUT_IN_SECONDS=25` 
and I don't get a timeout.error anymore!
> mhm, ok, then let's
> 
> * check if there is a better way to communicate errors in Rasa to the `console` connector than the timeout
> * if not: make the timeout configurable

But my big question is can I adjust this default stream error time on RASA-X as well? @wochinge 

Because I really need to share my bot for my Masterthesis asap and I think it's the last step, from prevting me to do so!

**Comment by wochinge** [comment]
> But my big question is can I adjust this default stream error time on RASA-X as well? @wochinge 

Rasa X communicates with Rasa Open Source in a different way, so you should be good

> Did you already do that?

No, we haven't done it yet.

**Comment by nbeuchat** [comment]
Just bumped into the same issue with `rasa==1.9.5`. I just wanted to use the console locally for testing (we usually only use the Facebook channel). 

I had to override `DEFAULT_STREAM_READING_TIMEOUT_IN_SECONDS` in the rasa code itself (which is ok locally but obviously not good).

It would be fantastic to be able to set `DEFAULT_STREAM_READING_TIMEOUT_IN_SECONDS` (or any other constant for that matter) through environment variable. I could have then just run:

`DEFAULT_STREAM_READING_TIMEOUT_IN_SECONDS=30 rasa shell`

**Comment by IgNoRaNt23** [comment]
Got the same problem when I start a bot in rest-mode.

**Comment by wochinge** [comment]
I made a PR to fix this: https://github.com/RasaHQ/rasa/pull/5686

**Comment by ngikgm** [comment]
i tried to look  at #5686 but still facing the same error. Will you be able to explain how to configure the time. Tried to amend the code in console.py .  Rasa shell command " --response-timeout " does not seems to work and return (rasa: error: unrecognized arguments: --response-timeout & rasa shell: error: invalid choice: '3600' (choose from 'nlu'))
![image](https://user-images.githubusercontent.com/56438262/85990727-0ba2db00-ba25-11ea-9915-3d9e0d742848.png)
